### ISSUE TITLE
not allowed statement in the FORALL construct

### ISSUE THREAD

**Issue by clementval** [omni-compiler/omni-compiler#164]
The following code triggers an error. 

```fortran
MODULE mod1

IMPLICIT NONE

TYPE type1
  LOGICAL, POINTER :: logical_array(:)
END TYPE type1

CONTAINS

  SUBROUTINE sub1(typ)
    TYPE (type1), INTENT(INOUT) :: typ
    INTEGER :: ierrstat, i
    INTEGER :: num, num_compute, io
    
    ALLOCATE(typ%logical_array(0:(num_compute - 1)), stat=ierrstat)
    FORALL(i = 0:(num_compute - 1))
      typ%logical_array(i) = (MOD(i, num) /= io)
    END FORALL
    
  END SUBROUTINE sub1
END MODULE mod1
```

```bash
"forall.f90:mod1", line 18: not allowed statement in the FORALL construct
```

**Comment by clementval** [comment]
@h-murai @shingo-s  The error is triggered because a statement in the list (`F-compile.c` #8573) as the code `FIRST_EXECUTION_POINT`. What is this kind of statement?

**Comment by shingo-s** [comment]
@clementval 

The comment in C-exprcode.def says:
```
# for FORTRAN Xcode

L    FIRST_EXECUTION_POINT

```
I'm not sure that FIRST_EXECUTION_POINT is used in the current code.
But it is maybe used for this old intermediate form.

Anyway, I'll fixed it with https://github.com/omni-compiler/omni-compiler/tree/bugfix/164.

**Comment by h-murai** [comment]
@clementval @shingo-s @mitsuhisa-sato I guess that FIRST_EXECUTION_POINT might be used as a placeholder for the border between the declaration part and the execution part.
